lxd/cluster/connect.go

```diff
@@ -50,23 +53,22 @@ func Connect(address string, networkCert *shared.CertInfo, serverCert *shared.Ce
 		args.UserAgent = clusterRequest.UserAgentNotifier
 	}
 
-	if r != nil {
+	if request.IsRequestContext(reqContext) {
 		proxy := func(req *http.Request) (*url.URL, error) {
```

Comment:
BTW @MusicDin @markylaing I find this entire proxy logic inside `cluster.Connect()` a bit suspect.

Why would a proxy need to be used for intra-cluster communication? And is this same proxy logic used for dqlite setup?

Reply:
I think this `proxy` logic is also used as a way to modify the request to send the request username and other auth details across to the cluster member. Not really to do with proxying.